### PR TITLE
[default.nix] Enable parallel build

### PR DESCRIPTION
**Pull request by vbgl**
This should reduce the total build time.

### CODE REVIEW

**Comment by ejgallego** [comment]
Hard to compare if build times are affected as the nix job doesn't print the needed worker information.

**Comment by vbgl** [comment]
What do you mean? Some hint would be welcome on how to address this issue.

**Comment by ejgallego** [comment]
> What do you mean? Some hint would be welcome on how to address this issue.

Oh sorry, workers have different numbers of processors and memory, what we do in the standard gitlab job is:
```
  - cat /proc/{cpu,mem}info || true
```
you need that in order to understand build times with -j

File: default.nix
Comment (Zimmi48) [inline default.nix]: I don't like having this here, it isn't related to the actual build. Please move it to `.gitlab-ci.yml`

**Review by Zimmi48 — COMMENTED** [review]
That's not the first time I'm explaining this but the reason for not enabling parallel building was because we build both `byte` and `world` targets and this could lead to races. The plan was to fix this by moving to Dune. With ongoing work by @ejgallego to build and install the HTML doc with dune, that should be possible soon.

**Comment by vbgl** [comment]
I don’t remember ever observing such a race. If that’s an issue, then this should not be merged. No problem.

**Comment by ejgallego** [comment]
> I don’t remember ever observing such a race. If that’s an issue, then this should not be merged. No problem.

Indeed there is a problem in the makefile so you need to call make twice.

**Comment by ejgallego** [comment]
I tried like 5 times to get a worker with 2 processors, no luck today... So I guess I leave this up to the Nix maintainers.

**Review by ejgallego — APPROVED** [review]

**Comment by Zimmi48** [comment]
We'll soon be able to switch the build to Dune anyways.

**Comment by ejgallego** [comment]
We finally got a builder with more than one core https://gitlab.com/coq/coq/-/jobs/179101249 , it seems to sped the build fine, merging.

**Comment by ejgallego** [comment]
Well, @Zimmi48 , I see you didn't green tick this, so I won't assign / merge until I get your approval.

**Comment by Zimmi48** [comment]
> We finally got a builder with more than one core https://gitlab.com/coq/coq/-/jobs/179101249 , it seems to sped the build up fine, merging.

We would need a few more similar runs to be sure there is no race. I prefer the solution of #9689 anyway.

**Comment by ejgallego** [comment]
> We would need a few more similar runs to be sure there is no race. I prefer the solution of #9689 anyway.

Ok, then let's close this I guess.

**Comment by vbgl** [comment]
If #9689 is for 8.11, then maybe this one is good for 8.10…

**Review by Zimmi48 — APPROVED** [review]
OK well, we can still merge and see if problems start occurring. If it's the case, it will be easy to revert.

**Comment by Zimmi48** [comment]
The Nix failure is due to an unreliable test in the test-suite that has been removed since then.

**Pull request merged**
